client/blocks/upload-image/README.md

```diff
@@ -0,0 +1,42 @@
+Upload Image
+=========
+
+`UploadImage` component is used to accommodate the following flow:
+- click on a designated place to open file explorer
+- in file explorer, select an image to upload
+- image is moved to `ImageEditor` where user can edit the image before uploading
+- after clicking on "Done", `UploadImage` calls the `onImageEditorDone` handler where you can perform any operation
+ (such as actual uploading of the image).
+ 
+The initial idea to create the block comes from the `edit-gravatar` one -- it's very similar but reusable and extensible.
+
+
+#### Basic usage:
+
+```js
+import UploadImage from 'blocks/upload-image';
+
+render: function() {
+	return
+		<UploadImage onImageEditorDone={ ( imageBlob ) => console.log( URL.createObjectURL( imageBlob ) ) } />;
+}
+```
+
+To see a more complex example, have a look at `blocks/upload-image/docs/example`.
+
+#### Props
+
+- `isUploading`: (default: false) whether to display a spinner over selected-n-edited image (use only after user 
+	selects an image in `onImageEditorDone`).
+- `imageEditorProps`: (default: allowedAspectRatios set to 1X1) object of additional props to send to `ImageEditor`
+	component.
+- `texts`: (default: `{}`) object of different texts/messages to show.
```

Comment:
Yes, `labels` over `texts`. Though I do  see that one of the props has a postfix of text. 

Alternatively instead of passing through a single object, we can break this apart as individual props to pass through. For example: `doneButtonText` and `addImageText`. I think I like this alternative better since we don't need to guard against accessing properties of null objects, it's clearer to the reader when looking at the component, and we can set prop defaults easily.